llvm/test/CodeGen/RISCV/rvv/vwadd-sdnode.ll

```diff
@@ -1466,3 +1466,159 @@ define <vscale x 2 x i32> @vwadd_wv_disjoint_or(<vscale x 2 x i32> %x.i32, <vsca
   %or = or disjoint <vscale x 2 x i32> %x.i32, %y.i32
   ret <vscale x 2 x i32> %or
 }
+
+define <vscale x 8 x i64> @vwadd_vx_splat_zext(<vscale x 8 x i32> %va, i32 %b) {
```

Comment:
I think these test cases where we're extending to the next power of two will already generate some form of vwadd via the tablegen patterns. But it looks like the codegen is slightly different. Do you think you could maybe add the tests in an earlier commit and rebase so we can see the diff.